src/lib.rs

```diff
@@ -464,24 +484,18 @@ fn pan_orbit_camera(
                 pan_orbit.target_focus += translation;
                 has_moved = true;
             }
-        } else if scroll.abs() > 0.0 {
-            let apply_scroll = |value: f32| value - scroll * value * 0.2;
-
-            if let Projection::Orthographic(ref mut p) = *projection {
-                p.scale = util::apply_zoom_limits(
-                    apply_scroll(p.scale),
-                    pan_orbit.zoom_upper_limit,
-                    pan_orbit.zoom_lower_limit,
-                );
-            } else {
-                pan_orbit.radius = pan_orbit.radius.map(|radius| {
-                    util::apply_zoom_limits(
-                        apply_scroll(radius),
-                        pan_orbit.zoom_upper_limit,
-                        pan_orbit.zoom_lower_limit,
-                    )
-                });
-            }
+        } else if (scroll_line + scroll_pixel).abs() > 0.0 {
+            // Calculate the amount of change based on line scroll and pixel scroll
+            let line_delta = -scroll_line * pan_orbit.target_radius * 0.2;
+            let pixel_delta = -scroll_pixel * pan_orbit.target_radius * 0.2;
+            // Add the calculated deltas to the target radius
+            pan_orbit.target_radius = util::apply_zoom_limits(
+                pan_orbit.target_radius + line_delta + pixel_delta,
+                pan_orbit.zoom_upper_limit,
+                pan_orbit.zoom_lower_limit,
+            );
+            // If it is pixel-based scrolling, it is added directly to the target
+            pan_orbit.radius = pan_orbit.radius.map(|radius| radius + pixel_delta);
```

Comment:
Going to comment on the new PR now.

You aren't applying zoom limits to this line - pixel scroll delta is being applied directly to radius without limits.

Reply:
> Going to comment on the new PR now.
> 
> You aren't applying zoom limits to this line - pixel scroll delta is being applied directly to radius without limits.

Perhaps we could add another function, like this one?
```
pub(super) fn interpolate_and_check_approx_f32(target: f32, current: f32, smoothness: f32) -> f32 {
    let t = 1.0 - smoothness;
    let mut new_value = current.lerp(&target, &t);
    if approx_equal(new_value, target) {
        new_value = target;
    }
    new_value
}
```